### PR TITLE
issue #195: Implement new mutect2 variant calling strategy

### LINKED ISSUE #195: Implement new mutect2 variant calling strategy
### Issue
Update Mutect2 variant calling to include latest gatk (4.2.0.0) recommended best practices.

### Approach

- Update mutect2 template
- Add or modify existing templates to include post mutect2 clean up templates (executed in this order)
  - GetPileupSummaries
  - CalculateContamination
  - LearnReadOrientation
  - SortBamout
  - **FilterMutectCalls** required step for downstream consensus variant calling
  - FilterAlignmentArtifacts

Related to this change, additional germline references are required (many are seen in mutect1 list, however not all are listed)
- Sequence Dictionary: seq dict of reference
- BWA img: .img file from this tool [https://gatk.broadinstitute.org/hc/en-us/articles/360040095652-BwaMemIndexImageCreator](url)
- Panel of normal
- Germline resource file
- Contamination VCF
For more information, please refer to this [tutorial](https://gatk.broadinstitute.org/hc/en-us/articles/360047232772--Notebook-Intro-to-using-Mutect2-for-somatic-data). 

### Important Points

- One important point for this new approach is the lack of germline resource file for mouse (mm10) genomes, meaning only FilterMutectCalls should be run for mouse genomes. This is required for somaticseq implementation because the vcf headers are changed by this tool.
- Realignment should also NOT BE PERFORMED for gatk4 mutect2. The realignment step has been internalized as described by authors. So for example in PostAlignment stage, AmpliconGatkRealign or AbraRealign should not be applied to mutect2 samples.

### PR DESCRIPTION
**Pull request by kamyshova**
Resolves #195
The pull request provides implementation of the new gatk4 mutect2 variant calling strategy:

- Added new templates after mutect2 tool:
  - GetPileupSummaries
  - CalculateContamination
  - LearnReadOrientation
  - SortBamout
  - FilterMutectCalls
  - FilterAlignmentArtifacts
- Realignment was skipped for gatk4 mutect2

### CODE REVIEW

**Comment by syansanofi** [comment]
@kamyshova 
**Different issues:**
I am getting -normal NA when using tumor only.   
Vardict & strelka2 all run without matching control with "NA" set as control.

**Comment by kamyshova** [comment]
@syansanofi Fixed

File: src/main/java/com/epam/fonda/tools/impl/FilterMutectCalls.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    private ToolFields initToolFields(final Configuration configuration) {
+        final GlobalConfig.DatabaseConfig databaseConfig = configuration.getGlobalConfig().getDatabaseConfig();
+        final GlobalConfig.ToolConfig toolConfig = configuration.getGlobalConfig().getToolConfig();
+        return ToolFields.builder()
+                .genome(validate(databaseConfig.getGenome(), GlobalConfigFormat.GENOME))
+                .gatk(validate(toolConfig.getGatk(), GlobalConfigFormat.GATK))
+                .vcf(vcf)
+                .contamTable(contamTable)
```
Comment (syansanofi) [inline src/main/java/com/epam/fonda/tools/impl/FilterMutectCalls.java:89]: Should be sample specific contamination table

`.contamTable(format("%s/%s.contamination.table", outputDir, sampleName))`

**Review by syansanofi — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/fonda/tools/impl/FilterMutectCalls.java (hunk 1 above)
Comment (kamyshova) [reply src/main/java/com/epam/fonda/tools/impl/FilterMutectCalls.java:89]: Fixed `CalculateContaminationOutput` in `CalculateContamination` tool

**Review by syansanofi — APPROVED** [review]

**Pull request merged**